pandas/_libs/tslibs/offsets.pyx

```diff
@@ -314,7 +314,9 @@ cdef _determine_offset(kwds):
 
     use_relativedelta = False
     if len(kwds_no_nanos) > 0:
-        if any(k in _kwds_use_relativedelta for k in kwds_no_nanos):
+        if "milliseconds" in kwds_no_nanos:
+            offset = timedelta(**kwds_no_nanos)
```

Comment:
I think this would still fail if a user passed `DateOffset(milliseconds=1, month=1)` i.e. any non-timedelta keyword constructor was passed

Reply:
Yeah it fails. What should be done in cases like this? Both timedelta and relativedelta will throw an error.

TBH I don't know how to fix this bug without trying to rework the entire function. Does it make sense to broaden up the scope of this issue?

Reply:
I think it would be okay to broaden the scope if all the existing tests pass. Generally I would imagine there should be checking that `kwds` (besides nanoseconds) should either all be valid for `timedelta` or `relativetimedelta`

Reply:
[This ](https://github.com/markopacak/pandas/blob/65015c80daca4812f8f097a60f843d6dd53d9487/pandas/_libs/tslibs/offsets.pyx#L300-L342) is what I had in mind.

All tests in `tests/arithmetic/test_datetime64.py` and `tests/tseries/offsets/test_offsets.py` pass.

[Added one parametrized test](https://github.com/markopacak/pandas/blob/ef7f11884383b59fbb000a3f50597e971d292932/pandas/_libs/tslibs/offsets.pyx#L300-L341). 

`DateOffset(milliseconds=1, month=1)` fails because I don't know what should happen in this case (see last return stmt in first link)

Reply:
Should I just open another PR with the rework linked above and close this one?

Reply:
I think it would be fine to just update this PR with your proposed refactor

Reply:
could we do something like

```
if 'milliseconds' in kwds:
     if "microseconds in kwds:
           raise ...
     kwds['microseconds'] = kwds.pop('milliseconds') * 1000
```

(and something similar in the `if "millisecond" in kwds_no_nanos` case below)

Reply:
I added this code instead of the exception block ([full code here](https://github.com/markopacak/pandas/blob/fcad617891a42d641e612ff64c130ed0222aaaf9/pandas/_libs/tslibs/offsets.pyx#L313)) :

```
if "millisecond" in kwds_no_nanos:
    micro = kwds_no_nanos.pop("millisecond") * 1000
    kwds_no_nanos["microsecond"] = kwds_no_nanos.get("microsecond", 0) + micro

if "milliseconds" in kwds_no_nanos:
    micro = kwds_no_nanos.pop("milliseconds") * 1000
    kwds_no_nanos["microseconds"] = kwds_no_nanos.get("microseconds", 0) + micro
```

but it breaks 2 tests in pandas/tests/tseries/offsets/test_offsets.py, both of which are meant to intentionally fail if we pass `millisecond`. So we aren't breaking any real functionality, I guess.

```
TestDateOffset.test_constructor[millisecond] 
[XPASS(strict)] Constructing DateOffset object with `millisecond` is not yet supported.

test_valid_relativedelta_kwargs[millisecond] 
[XPASS(strict)] Constructing DateOffset object with `millisecond` is not yet supported.
```
However, I think the requirement is to have all the existing tests pass.


There is still the case where we receive `minutes` (arg valid for `timedelta`)  + another arg valid for `relativedelta` only. 
In general, the last return needs to be discussed: What do we do in "all other cases"?